src/components/core/GeoChart.tsx

```diff
@@ -0,0 +1,126 @@
+// (C) 2019-2020 GoodData Corporation
+import * as React from "react";
+import * as cx from "classnames";
+import { WrappedComponentProps } from "react-intl";
+
+import { Execution } from "@gooddata/typings";
+
+import { ICommonChartProps } from "./base/BaseChart";
+import { BaseVisualization } from "./base/BaseVisualization";
+import { commonDefaultProps, ILoadingInjectedProps } from "./base/VisualizationLoadingHOC";
+import { createPushPinDataLayer } from "./geoChart/geoChartDataLayers";
+import { createPushPinDataSource } from "./geoChart/geoChartDataSource";
+import GeoChartLegendRenderer, { IChartLegendProps } from "./geoChart/GeoChartLegendRenderer";
+import GeoChartRenderer, { IChartProps } from "./geoChart/GeoChartRenderer";
+
+import { IDataSourceProviderInjectedProps } from "../afm/DataSourceProvider";
+import { LoadingComponent } from "../simple/LoadingComponent";
+import { DEFAULT_DATA_SOURCE_NAME } from "../../constants/geoChart";
+import { IChartConfig } from "../../interfaces/Config";
+import { getGeoDataIndex } from "../../helpers/geoChart";
+
+export interface IChartHTMLElement extends HTMLDivElement {
+    getChart(): mapboxgl.Map;
+    getChartRef(): HTMLDivElement;
+}
+
+export function renderChart(props: IChartProps): React.ReactElement {
+    return <GeoChartRenderer {...props} />;
+}
+
+export function renderLegend(props: IChartLegendProps): React.ReactElement {
+    return <GeoChartLegendRenderer {...props} />;
+}
+
+export interface IGeoChartProps extends ICommonChartProps, IDataSourceProviderInjectedProps {
+    config?: IChartConfig;
+    execution: Execution.IExecutionResponses;
+    chartRenderer(props: IChartProps): React.ReactElement;
+    legendRenderer(props: IChartLegendProps): React.ReactElement;
+}
+
+export type IGeoChartInnerProps = IGeoChartProps &
+    ILoadingInjectedProps &
+    IDataSourceProviderInjectedProps &
+    WrappedComponentProps;
+
+/**
+ * Geo Chart react component
+ */
+export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
+    public static defaultProps: Partial<IGeoChartInnerProps> = {
+        ...commonDefaultProps,
+        chartRenderer: renderChart,
+        legendRenderer: renderLegend,
+    };
+
+    private chartRef: IChartHTMLElement;
+
+    public renderVisualization() {
+        const classes = cx("gd-geo-component");
+        return (
+            <div className={classes}>
```

Comment:
I don't think we need `const classes = cx("gd-geo-component");`
It could be `<div className="gd-geo-component">`